.github/workflows/pull-request.yaml

```
@@ -1,11 +1,212 @@
 name: Test Changes
 
-on: [pull_request]
+on:
+  pull_request:
+    paths-ignore:
+      - '*.md'
+
+  push:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+.x
+
+env:
+  GRADLE_OPTS: '-Dorg.gradle.daemon=false -Dkotlin.compiler.execution.strategy=in-process'
 
 jobs:
   build:
     runs-on: ubuntu-18.04
 
     steps:
-      - name: Dummy job
-        run: echo "Hello world"
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Cache gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ~/.konan
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Install Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Build project and run tests
+        run: ./gradlew build --stacktrace
+
+      - name: Cache protobuf
+        uses: actions/cache@v2
+        id: cache-protobuf
+        with:
+          path: ~/protobuf
+          key: ${{ runner.os }}-protobuf-3.10.1
+
+      - name: Build protoc and protobuf conformance-test-runner
+        if: steps.cache-protobuf.outputs.cache-hit != 'true'
+        run: |
+          cd ~
+          curl -sSLO https://github.com/protocolbuffers/protobuf/releases/download/v3.10.1/protobuf-all-3.10.1.tar.gz
+          tar xzvf protobuf-all-3.10.1.tar.gz
+          mv protobuf-3.10.1 protobuf
+          cd protobuf
+          ./configure --prefix="$(pwd)/install"
+          make
+          make install
+          cd conformance
+          make
+
+      - name: Ensure bundled types are up-to-date
+        run: |
+          export PATH="$PATH:$HOME/protobuf/install/bin"
+          ./gradlew -Dprotoc.path=$HOME/protobuf/install \
+            :runtime:generateWellKnownTypes \
+            :runtime:generateTestTypes \
+            :protoc-gen-kotlin:protoc-gen-kotlin-lib:generateProto \
+            :conformance:conformance-lib:generateProto
+          if [ -n "$(git status --porcelain)" ]; then
+            echo "*** Found uncommitted changes to bundled types: ***"
```

It'd be helpful to make this an official error message, so GitHub will explain what went wrong on the run instead of `Exited with error code 1`

```suggestion
            echo "::error::Found uncommitted changes to bundled types"
```

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-error-message

---

Ooh, nice 👍